src/test/java/com/delivery/store/model/repository/StoreRepositoryTest.java

```diff
@@ -119,7 +118,7 @@ void update() {
     void delete() {
         // given
         Store store = storeRepository.findById(1L)
-            .orElse(null);
+            .orElseThrow(() -> new NoSuchElementException("해당 ID는 존재하지 않습니다."));
```

Comment:
해당 테스트에서는 delete 메소드를 실행해서 상태 값이 바뀌는지만 테스트하는 것이 목적이므로 에러를 던지도록 하는 것보다는
Optional get으로 값을 직접 가져와도 될 것 같습니다.

Reply:
repository에서 엔티티 추가, 생성이 entity에 @Column 어노테이션 설정을 통해 제약 조건을 위반 했는지 테스트도 가능합니다.

Reply:
음 UserRepositoryTest를 보자면 아래와 같이 작성하셨는데요
```suggestion
User savedUser = userRepository.findByEmail(userDto.getEmail()).get();
```
Optional이 의미하는 것은 Nullable하다는 것이기 때문에
직접적으로 get()을 가져오는 행위는 Optional의 anti pattern 입니다.
물론 비즈니스적으로 풀자면 get으로 받아와도 문제는 없습니다만
optional을 쓴다는 것은 다음과 같이 처리하는게 맞다고 생각합니다.

```
Optional<Store> optional = storeRepository.findById(1L);
if (optional.isPresent()) {
      Store store = optional.get();
}
```
다만, 이런식으로 사용하는 것은 가독성을 낮추기 때문에

```java
Optional<T>.orElseThrow()
Optional<T>.orElse()
```
이런식으로 푸는게 맞지 않을까 생각합니다.

https://dzone.com/articles/using-optional-correctly-is-not-optional

Reply:
service layer에서는 repostiry 호출 시 저렇게 사용해야 하지만 위 코드는 test에서 user의 해당 기능 만들 테스트 하기 위함이 목적이므로 optional의 값을 체크하여 없을 때 null 또는 exception을 넘겨주기보다는 해당 기능을 테스트하는게 맞는 것 같습니다.
null 또는 excpetion을 넘겨준다면 해당 테스트에서 그에 대응하는 코드 또한 넣어줘야 하구요

Reply:
값이 없을 때, exception class를 테스트하는 메소드가 따로 있다면 해당 기능에 대해 검증이 이뤄진것이므로, delete를 테스트하는 코드에서는 exception을 넣지 않아도 될 것 같습니다.